src/client/consumer.ts

```diff
@@ -70,29 +74,89 @@ class Consumer implements vscode.Disposable {
      * Received messages and/or errors are emitted via events.
      */
     async start(): Promise<void> {
+        const partitions = this.options.partitions;
+        const partitionAssigner = this.getPartitionAssigner(partitions);
+        const fromOffset = this.options.fromOffset;
+        const topic = this.options.topicId;
+
         this.kafkaClient = createKafka(this.options);
-        this.consumer = this.kafkaClient.consumer({ groupId: `vscode-kafka-${this.options.clusterId}-${this.options.topic}`, retry: { retries: 3 }});
+        this.consumer = this.kafkaClient.consumer({
+            groupId: this.options.consumerGroupId, retry: { retries: 3 },
+            partitionAssigners: [
+                partitionAssigner
+            ]
+        });
         await this.consumer.connect();
-        await this.consumer.subscribe({ topic: this.options.topic, fromBeginning: this.options.fromOffset ===  "earliest" })
 
-        await this.consumer.run({
-            eachMessage: async ({ topic , partition, message }) => {
+        const subscribeOptions = this.createSubscribeOptions(topic, fromOffset);
+        await this.consumer.subscribe(subscribeOptions)
+
+        this.consumer.run({
+            eachMessage: async ({ topic, partition, message }) => {
                 this.onDidReceiveMessageEmitter.fire({
                     uri: this.uri,
                     record: { topic: topic, partition: partition, ...message },
                 });
             },
         });
-    }
 
-    private parseUri(uri: vscode.Uri): { clusterId: string; topic: string; partition?: string} {
-        const [clusterId, topic, partition] = uri.path.split("/");
+        if (partitions || (fromOffset && subscribeOptions.fromBeginning == undefined)) {
+            const definedPartitions = partitions || [0];
+            for (let i = 0; i < definedPartitions.length; i++) {
+                const partition = definedPartitions[i];
+                this.consumer.seek({ topic, partition, offset: fromOffset || '0' });
+            }
+        }
+    }
 
-        return {
-            clusterId,
-            topic,
-            partition,
-        };
+    private getPartitionAssigner(partitions?: number[]) : PartitionAssigner {
+        if (partitions) {
+            const userData = Buffer.alloc(0);
+            return ({ cluster }) => ({
+                name: 'AssignedPartitionsAssigner',
+                version: 1,
+                async assign({ members, topics }) {
+                    const sortedMembers = members.map(({ memberId }) => memberId).sort()
```

Comment:
I'm thinking if we can simplify this a bit. In the syntax we only allow a single topic, and we already assume a single member (the extension instance) (which leads me to think if we in the future might want to allow this to join any other consumer group - but let's leave that out for now).

I'm not too familiar with the custom assigner in kafka.js, but would something like this work?


```
Can we simplify this assigner?
I'm thinking if we can assume a single consumer member then something like this might be doable:

```
const assignment = Object.keys(topics).map(topic => ({ topic: partitions }); 
```

Reply:
I'm sorry, there is an error syntax with your suggestion.